File: apps/sdkwork-appstore-mini-program/packages/sdkwork-appstore-mp-collection/src/services/collectionService.ts
import type { AppstoreAppClient } from "@sdkwork/appstore-mp-core";

import type { CollectionPageResult } from "../types/collectionModels.js";

/**
 * Collection service.
 *
 * The generated App Store app SDK client is injected by the root bootstrap
 * (`APP_SDK_INTEGRATION_SPEC.md`); this package never constructs a client and
 * never issues raw HTTP.
 */
export interface CollectionService {
  readonly capability: "collection";
}

export function createCollectionService(client: AppstoreAppClient): CollectionService {
  void client;
  return { capability: "collection" };
}

export function emptyCollectionResult(): CollectionPageResult<never> {
  return { items: [] };
}

File: apps/sdkwork-appstore-flutter-mobile/packages/sdkwork_appstore_flutter_mobile_collection/lib/src/services/collection_service.dart
import 'package:sdkwork_appstore_flutter_mobile_core/sdkwork_appstore_flutter_mobile_core.dart';

import '../models/collection_models.dart';

/// Collection service.
///
/// The App Store app SDK clients are injected by the root bootstrap
/// (`APP_SDK_INTEGRATION_SPEC.md`); this package never constructs a client and
/// never issues raw HTTP.
class CollectionService {
  const CollectionService({required this.clients});

  final AppstoreAppSdkClients clients;

  String get capability => 'collection';

  CollectionPageResult<Never> empty() =>
      const CollectionPageResult<Never>(items: <Never>[]);
}
